### PR TITLE
Retries around command script upload and run creation.  Closes #808

### LINKED ISSUE #808: Add retries around command script and JES run creation
See #797

### PR DESCRIPTION
**Pull request by mcovarr**
Wheel favorites @geoffjentry and @cjllanwarne

### CODE REVIEW

**Comment by geoffjentry** [comment]
imgres

👍

**Comment by cjllanwarne** [comment]
What Jeff said. 👍

File: supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesAsyncBackendJobExecutionActor.scala [hunk 1]
```diff
@@ -310,15 +322,23 @@ class JesAsyncBackendJobExecutionActor(override val jobDescriptor: BackendJobDes
          |echo $$? > $rcPath
        """.stripMargin.trim
 
-    Try(gcsExecPath.write(fileContent))
+    val writeScript: Future[Unit] = Future(gcsExecPath.write(fileContent))
+
+    implicit val system = context.system
+    Retry.withRetry(
+      writeScript,
```
Comment (cjllanwarne) [inline supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesAsyncBackendJobExecutionActor.scala]: Wait a second! This is a Future not a thunk!

File: core/src/main/scala/cromwell/core/retry/Retry.scala [hunk 2]
```diff
@@ -13,7 +13,7 @@ object Retry {
     * Retries a Future on a designated backoff strategy until either a designated number of retries or a fatal error
     * is reached.
     *
-    * @param f A Future thunk which will be executed once per cycle
+    * @param f A function Unit => Future which will be executed once per cycle.
```
Comment (geoffjentry) [inline core/src/main/scala/cromwell/core/retry/Retry.scala]: still a thunk although this might be less confusing anyways

**Pull request merged**